src/Contracts/Str.php

```diff
@@ -0,0 +1,46 @@
+<?php
```

Comment:
I think we should support both, as suggested [here](https://github.com/phpol/phpol/issues/3#issuecomment-277754447) but reinforce the immutable state by suffixing the class.